core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,369 @@
+    maf = maf[headers]
+    # Remap maf file using important parts, as cbioportal MAF format does not conform to
+    #  GFC MAF file format (column "Consequence" is not placed on the correct location).
+    new_headers = ["NA" for i in range(16)]
+    for i, j in enumerate([4, 5, 6, 10, 12, 15]):
+        new_headers[j] = headers[i]
+    maf = maf.reindex(new_headers, axis=1)
+    # Fill empty cells with NA string, otherwise they get parsed out by SigProfilerMatrixGenerator
+    maf = maf.fillna("NA")
+    # File needs the MAF file extension, otherwise it is not recognized by SigProfilerMatrixGenerator
+    outfile = tmp_dir + "/mutations.maf"
+    maf.to_csv(outfile, sep="\t", index=False)
+    return outfile, genome
+
+
+def genome_install(genome):
+    """Install a genome for SigProfilerMatrixGenerator
+    
+    :param genome: genome build (GRCh37 or GRCh38)
+    :return: None    
+    """
+    genInstall.install(genome)
+    return None
+
+
+def run_matrix_generator(genome_build, maf_folder):
+    """Run SigProfilerMatrixGenerator to generate substitution matrices in input folder
+    
+    :param genome_build: str, genome build (GRCh37 or GRCh38)
+    :param maf_folder: str, path to folder containing maf input files
+    :return: None
+    """
+    print("-" * 60 + "\nGenerating nucleotide matrices: ")
+    # Name of the project is irrelevant, these files will be removed after MS algorithm has been run
+    project = "tmp"
+    matGen.SigProfilerMatrixGeneratorFunc(project, genome_build, maf_folder)
+    return None
+
+
+def install_temposig():
+    """Call R to install tempoSig
+    
+    :return: None
+    """
+    robjects.r('devtools::install_github("mskcc/tempoSig")')
+    return None
+
+
+def check_install_temposig():
+    """Call R to check if tempoSig is installed and where, otherwise install using R
+    
+    :return: str, location of tempoSig.R script
+    """
+    temposig_location = robjects.r("system.file('exec', 'tempoSig.R', package = 'tempoSig')")[0]
+    if not temposig_location:
+        install_temposig()
+        temposig_location = robjects.r("system.file('exec', 'tempoSig.R', package = 'tempoSig')")[0]
+        if not temposig_location:
+            raise ImportError("Unable to install or locate the tempoSig R-package. Please install manually. "
+                              "If this error persists, either use the --temposig parameter or use another algorithm.")
+    return temposig_location
+
+
+def matrix_temposig(file_name):
+    """Remove the first column on the first row, tempoSig requires this
+
+    :param file_name: str, name of matrix file to alter
+    :return: None, file is altered on host machine
+    """
+    with open(file_name, "r") as infile:
+        data = infile.readlines()
+    data[0] = "\t".join(data[0].split("\t")[1:])
+    with open(file_name, "w") as outfile:
+        outfile.writelines(data)
+    return None
+
+
+def run_temposig(temposig_loc, inf_matrix, outf, cosmic=2, nperm=1000):
```

Comment:
instead of `cosmic` i'd suggest to call it `version`

Reply:
I'll rename it to cosmic_version, as not to confuse it with the TempoSig version.